src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -293,10 +293,36 @@ struct patcher
   }
 
   void
-  patch_it(uint8_t* base, uint64_t new_value)
+  patch_it(uint8_t* base, uint64_t value)
   {
+    patch_it_impl(base, value);
+  }
+
+  void
+  patch_it(xrt::bo bo, uint64_t value)
+  {
+    patch_it_impl(bo, value);
+  }
+
+private:
+  template<typename T>
+  void
+  patch_it_impl(T base_or_bo, uint64_t new_value)
+  {
```

Comment:
I would recommend to keep just the uint8_t version, then do the map & cast where you are calling patch_it_impl.
What you have here, is not the right use of a template.
```
void
patch_it_impl(uint8_t* base, uint64_t new_value)
{
   ...
}
```